src/Sulu/Component/SmartContent/Orm/BaseDataProvider.php

```diff
@@ -150,28 +149,29 @@ public function resolveResourceItems(
 
     /**
      * Resolves filters.
+     *
+     * @param null $limit
+     * @param int $page
+     * @param null $pageSize
+     * @param array $options
+     *
+     * @return array
      */
     private function resolveFilters(
         array $filters,
-        $locale,
+        string $locale,
         $limit = null,
         $page = 1,
         $pageSize = null,
         $options = []
     ) {
-        $webspace = $this->requestAnalyzer ? $this->requestAnalyzer->getWebspace() : null;
-
         $result = $this->repository->findByFilters(
             $filters,
             $page,
             $pageSize,
             $limit,
             $locale,
-            $options,
-            $this->security && $webspace && $webspace->hasWebsiteSecurity() ? $this->security->getUser() : null,
```

Comment:
This can not be removed , it's used in https://github.com/sulu/sulu/blob/2.x/src/Sulu/Bundle/MediaBundle/Entity/MediaDataProviderRepository.php#L79 .
Maybe you can add it this two parameters to interface but i think it's a BC Break